src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/CreateImageIndexTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using FakeItEasy;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.NET.Build.Containers.IntegrationTests;
+using Microsoft.NET.Build.Containers.UnitTests;
+using NuGet.Protocol;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.NET.Build.Containers.Tasks.IntegrationTests;
+
+[Collection("Docker tests")]
+public class CreateImageIndexTests
+{
+    private ITestOutputHelper _testOutput;
+
+    public CreateImageIndexTests(ITestOutputHelper testOutput)
+    {
+        _testOutput = testOutput;
+    }
+
+    [DockerAvailableFact]
+    public async Task CreateImageIndex_Baseline()
```

Comment:
This test covers the fact that we can use the CreateImageIndex Task to create a manifest, but it does the 'orchestration' manually. Can you 

* add a test that that just calls the 'normal' entrypoint and verifies that the multi-rid orchestration does happen for publishing the index to a registry?
* add a test that that just calls the 'normal' entrypoint and verifies that the multi-rid orchestration does happen for publishing to tarballs?

These are the two major use cases that our partner team has.

Reply:
Added end to end tests EndToEndMultiArch_LocalRegistry, EndToEndMultiArch_ArchivePublishing, EndToEndMultiArch_Registry